### PR TITLE
[SQL] Add managed instances external governance status update

### PR DESCRIPTION
**Pull request by vahalik**
## Description

* Added a new cmdlet for Azure SQL Managed Instance refresh external governance status
  - 'Invoke-AzSqlInstanceExternalGovernanceStatusRefresh'
* Updated `Get-AzSqlInstance` to support returning the `ExternalGovernanceStatus` property

## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [X] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [X] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request
This pull request introduces changes that primarily focus on the implementation of a new feature for Azure SQL Managed Instances, specifically the ability to refresh the external governance status. The most significant changes include the addition of a new cmdlet, `Invoke-AzSqlInstanceExternalGovernanceStatusRefresh`, and the introduction of new classes and methods to support this functionality.

### CODE REVIEW

File: src/Sql/Sql/ManagedInstance/Cmdlet/AzureSqlInstanceRefreshExternalGovernanceCmdletBase.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Text;
+
+namespace Microsoft.Azure.Commands.Sql.ManagedInstance.Cmdlet
+{
+    public abstract class AzureSqlInstanceRefreshExternalGovernanceCmdletBase : AzureSqlCmdletBase<RefreshExternalGovernanceMIModel, RefreshExternalGovernanceAdapter>
+    {
+        /// <summary>
```
Comment (krivi37) [inline src/Sql/Sql/ManagedInstance/Cmdlet/AzureSqlInstanceRefreshExternalGovernanceCmdletBase.cs:28]: Please add parameter sets if you have the time. You can check cmdlets like [SetAzureSqlManagedInstanceDtc.cs](https://github.com/Azure/azure-powershell/blob/main/src/Sql/Sql/ManagedInstance/Cmdlet/SetAzureSqlManagedInstance.cs) for some parameter set examples, and also there is an example in the docs:
https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/examples/set-cmdlet-example.md
Basically, the 2 additional parameter sets would allow the users to pass resourceId to the cmdlet, or to pass an MI object (which would be fetched by Get/List cmdlet for example)

**Comment by krivi37** [comment]
Please add tests also

**Comment by wyunchi-ms** [comment]
Hi @vahalik , we have a best practices for managed identity [here](https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/design-guidelines/managed-identity-best-practices.md). Please take a look at it

**Comment by vahalik** [comment]
> Hi @vahalik , we have a best practices for managed identity [here](https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/design-guidelines/managed-identity-best-practices.md). Please take a look at it

Hi, thx for note. This PR is not about managed identity but managed instances under SQL.

**Comment by wyunchi-ms** [comment]
> > Hi @vahalik , we have a best practices for managed identity [here](https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/design-guidelines/managed-identity-best-practices.md). Please take a look at it
> 
> Hi, thx for note. This PR is not about managed identity but managed instances under SQL.

OK, Then I will edit the title to make less misunderstanding.

File: src/Sql/Sql/ManagedInstance/Cmdlet/InvokeAzSqlInstanceExternalGovernanceStatusRefresh.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Sql.ManagedInstance.Model;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+namespace Microsoft.Azure.Commands.Sql.ManagedInstance.Cmdlet
+{
+    [Cmdlet("Invoke", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlInstanceExternalGovernanceStatusRefresh", SupportsShouldProcess = true), OutputType(typeof(RefreshExternalGovernanceMIModel))]
+    public class InvokeAzSqlInstanceExternalGovernanceStatusRefresh : AzureSqlInstanceRefreshExternalGovernanceCmdletBase
+    {
+        private const string SetByNameParameterSet = "SetByNameParameterSet";
```
Comment (krivi37) [inline src/Sql/Sql/ManagedInstance/Cmdlet/InvokeAzSqlInstanceExternalGovernanceStatusRefresh.cs:27]: Tiny nitpick: Use the word "Invoke" instead of "Set:" i.e. "InvokeByNameParameterSet"

File: src/Sql/Sql/help/Invoke-AzSqlInstanceExternalGovernanceStatusRefresh.md [hunk 3]
```diff
@@ -12,9 +12,25 @@ Refreshes the value of external governance on the managed instance.
 
 ## SYNTAX
 
+### SetByNameParameterSet
```
Comment (krivi37) [inline src/Sql/Sql/help/Invoke-AzSqlInstanceExternalGovernanceStatusRefresh.md:15]: Here we should actually change to **Invoke**ByNameParameterSet because these examples are visible in documentation and in Help cmdlet

File: src/Sql/Sql.Test/ScenarioTests/RefreshSqlInstanceExternalGovernanceCmdletTests.ps1 [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+		# ------------------------------ Test by passing in the instance object
+
+		Write-Debug "Test by passing in the instance object"
+		$result = Get-AzSqlInstance -ResourceGroupName $rgName -Name $managedInstanceName | Invoke-AzSqlInstanceExternalGovernanceStatusRefresh
+
+		Write-Debug ('$result is ' + (ConvertTo-Json $result))
+		Assert-NotNull $result
+		
+		Assert-AreEqual $result.Status "Succeeded"
+		Assert-AreEqual $result.InstanceName $instanceName
```
Comment (krivi37) [inline src/Sql/Sql.Test/ScenarioTests/RefreshSqlInstanceExternalGovernanceCmdletTests.ps1:49]: I would also add 1 get call in these tests. Since we cannot enable external governance using the refresh cmdlet alone (it needs to be enabled on the Purview side, and I think that they don't have PowerShell cmdlets for now, just Portal and raw API requests), I would just get the MI using get cmdlet and check if the ExternalGovernanceStatus is set to False - just as a basic test which check whether the cmdlet actually returns that field

**Review by krivi37 — APPROVED** [review]
Looks good to me

**Review by wyunchi-ms — APPROVED** [review]
